source/_docs/composer-drupal-8.md

```diff
@@ -0,0 +1,130 @@
+
+Much of the information in this section comes from [drupal.org documentation on Composer usage](https://www.drupal.org/node/2404989).
+
+Composer assumes that packages use Semantic Versioning where releases have three numbers signifying `MAJOR.MINOR.PATCH` releases. Drupal modules however have release numbers like `8.x-1.2`. This pattern indicates that the release is the third official release (`8.x-1.0` and `8.x-1.1` being the first two) made from the `8.x-1.x` branch. The `8.x` indicates that the release is compatible with Drupal 8. All this to say, Drupal needs an alternative to packagist.org in order to map Drupal release numbers to Composer-compatible semantic versions.
+
+The first step to installing Drupal modules with Composer is to ensure that the `composer.json` in your Drupal site knows about an alternative to packagist.org where Drupal modules can be found. [Eventually, drupal.org will provide this information](https://www.drupal.org/node/2718229). For now, use https://packagist.drupal-composer.org.
+
+```
+composer config repositories.drupal composer https://packagist.drupal-composer.org
+```
+
+Once Composer knows where to find modules, they can be added easily with `composer require`. For instance, [address module](https://www.drupal.org/project/address) could be installed via Composer with
+
+```
+composer require drupal/address
+
+```
+
+Address module depends on a few external packages (because problems of mailing addresses are not Drupal-specific). Those external packages are downloaded to the `vendor` directory. Address module itself will be put in Drupal's `module` directory. `composer require` can also be used for themes and profiles which will also be placed in the appropriate Drupal directories.
+
+
+## Whether to commit the vendor directory
+
+[Much of the documentation on Composer says that the `vendor` directory should not be committed to version control.](https://getcomposer.org/doc/faqs/should-i-commit-the-dependencies-in-my-vendor-directory.md) This recommendation presumes some abstraction layer between the git repository developers commit to and the way their code is deployed to a live site (and test sites for that matter). That abstraction might be a build system that takes the development team's repository, and creates a "build artifact" from the repository (by running `composer install` and possibly other steps like compiling Sass) which is then deployed. That build artifact might be tracked in a separate git repository. [In fact, this mental model has been the way many teams have used Pantheon for years](https://pantheon.io/blog/example-repository-build-drupal-composer-travis). They use GitHub, Bitbucket, or some other external repository as the repository where developers commit their changes. A continuous integration service then builds the developers' changes and sends them to Pantheon. If this is the model you are using, you do not have to commit your `vendor` directory.
```

Comment:
@ataylorme, @dustinleblanc, could I get your thoughts on whether this section, or any other in the doc needs re-writing?